File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        '''
        Time Complexity: O(n) n is len(nums)
        Space Complexity: O(n)
        '''

        num_set = set(nums)
        ans = 0
        for num in nums:
            if num - 1 not in num_set:
                temp = 1
                while num + temp in num_set:
                    temp += 1
                ans = max(ans, temp)
        return ans

File: Data Structures & Algorithms/jump-game-ii/submission-4.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int jumps{0}, l{0}, r{0};

        while (r < nums.size() - 1) {
            int mdist{max(l + nums[l], r + nums[r])};
            l = r + 1;
            r = mdist;
            jumps++;
        }
        return jumps;
    }
};
